### ISSUE TITLE
Enhancement/Fix: Login with HTTP headers

### ISSUE THREAD

**Issue by 4o66** [wekan/wekan#4311]
### Server Setup Information

* Did you test in newest Wekan?: Yes
* Did you configure root-url correctly so Wekan cards open correctly (see https://github.com/wekan/wekan/wiki/Settings)? Yes
* Operating System: Ubuntu Server 21.04
* Deployment Method (Snap/Docker/Sandstorm/bundle/source): Docker
* Http frontend if any (Caddy, Nginx, Apache, see config examples from Wekan GitHub wiki first): Traefik/Authelia
* Node.js Version: Included in docker compose file
* MongoDB Version: Included in docker compose file
* What webbrowser version are you using (Wekan should work on all modern browsers that support Javascript)? Firefox 

#### Reproduction Steps
Logging in via HTTP headers has bad assumptions

https://github.com/wekan/wekan/commit/ff825d6123ecfd033ccb08ce97c11cefee676104

This code assumes that the uid and password as passed via the headers. Using Traefik/Authelia as an example, The authentication is performed via Authelia, and the headers contain only the uid, name, and email address. In this case Wekan should assume authentication is successful, and look up account using uid and/or email passed via header. For security, there should also be an option for trusted proxy IP, so that headers are only trusted from known server.

**Comment by xet7** [comment]
@4o66 

Please add pull request to fix it. AFAIK logging in via HTTP headers does not work yet.

**Comment by xet7** [comment]
Related #2019

**Comment by ocdtrekkie** [comment]
I would note that Sandstorm login is inherently "logging in with HTTP headers"... so there might be a possibility to simplify the Sandstorm integration here, I think?

**Comment by xet7** [comment]
@ocdtrekkie 

Yes, Sandstorm login is "logging in with HTTP headers", and it does work. I have not read it's code about how it works. I also don't know how complex is current Sandstorm integration? Well, I'll try to read some code.

That #2019 would be different case, there in front of WeKan is login proxy server  that prevents those headers from Internet, and in login process sends headers to WeKan. Well, when it sometime is fixed to work OK.

**Comment by xet7** [comment]
@4o66 

Headers for trusted iframe URL settings used to work, before browsers changes so that code does not work anymore. Some time ago I did see some related code that would work with new browsers, but I don't remember anymore what code it was.
https://github.com/wekan/wekan/blob/master/server/policy.js#L6
